### PR TITLE
add metadata for Twitter

### PR DESCRIPTION
**Pull request by maelle**
Regarding twitterImg if all goes well it works like twitterImg for roweb2 cf https://github.com/ropensci/roweb2/#blog-post-editorial-suggestions

### CODE REVIEW

**Comment by maelle** [comment]
wait a bit the summary doesn't work :-)

**Comment by maelle** [comment]
Ok now Twitter metadata only comes from head, but maybe it's not good to deviate from the original theme that much? At least the Twitter card should look ok for the base URL (site title and description, logo) and blog posts (post title and first sentences, logo).

File: config.toml [hunk 1]
```diff
@@ -27,4 +27,4 @@ lang = "en"
 github = "reside-ic"
 posts_navigation = true
 small_banner_logo = false
-latestpostcount = 5
\ No newline at end of file
+latestpostcount = 5
```
Comment (maelle) [inline config.toml:30]: not sure what I did here

Comment (richfitz) [reply config.toml:30]: the file did not previously have a trailing newline - you have opened this file in an editor which has added one - it's just noise :)

Comment (maelle) [reply config.toml:30]: should I remove it?

Comment (richfitz) [reply config.toml:30]: no, it's totally fine (my editor adds them too)

File: layouts/partials/head_includes.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    href="{{ .Site.BaseURL }}img/favicon.ico"
+{{ end }}
+>
+
+<!-- RSS -->
+{{ if .RSSLink }}
+  <link href="{{ .RSSLink }}" rel="alternate" type="application/rss+xml" title="{{ .Site.Title }}" />
+{{ end }}
+
+<!-- Twitter metadata -->
```
Comment (hillalex) [inline layouts/partials/head_includes.html]: Wouldn't it make more sense for this section to live in the `meta` partial below?

**Review by hillalex — COMMENTED** [review]
Thanks so much for doing this! I'm not too familiar with Hugo - so if you add your own files into the `layouts` directory they override the theme ones? Looks to me like the new meta tags could live in the meta partial - was there a reason for deviating from that?

File: layouts/partials/head_includes.html (hunk 2 above)
Comment (maelle) [reply layouts/partials/head_includes.html]: switched!

**Comment by maelle** [comment]
> so if you add your own files into the layouts directory they override the theme ones? 

Yes, and it seems to be recommended cf e.g. https://bookdown.org/yihui/blogdown/custom-layouts.html "a Hugo theme may be constantly updated by its original author for improvements or bug fixes. Similar to the “you break it, you buy it” policy (the Pottery Barn rule), once you touch someone else’s source code, you will be responsible for its future maintenance, and the original author should not be responsible for the changes you made on your side. That means it may not be easy to pull future updates of this theme to your website (you have to carefully read the changes and make sure they do not conflict with your changes), but if you are completely satisfied with the current state of the theme and do not want future updates, it is fine to modify the theme files directly."

> Looks to me like the new meta tags could live in the meta partial - was there a reason for deviating from that?

No, best to change that. I had actually not seen meta.html at first which was a headache :joy:

**Comment by hillalex** [comment]
Cool, good to know that's the recommended way to customise things, thanks again :smiley:

**Comment by maelle** [comment]
I think it's also clearer when you want to see what you customized (I haven't been consistent with this for my own blog, and now need to look up the commit history, less cool).

**Comment by maelle** [comment]
I'm waiting for the changes to deploy, I hope they'll look good on https://cards-dev.twitter.com/validator

**Pull request merged**